rDNA/R/rDNA.R

```diff
@@ -1806,4 +1807,289 @@ print.dna_barplot <- function(x, trim = 30, attr = TRUE, ...) {
   x2$Entity <- sapply(x2$Entity, function(e) if (nchar(e) > trim) paste0(substr(e, 1, trim - 1), "*") else e)
   class(x2) <- "data.frame"
   print(x2)
+}
+
+#' Plot dna_barplot object.
+#'
+#' Plot barplots generated from dna_barplot.
+#'
+#' This function plots agreement and disagreement towards DNA Statements for
+#' different entities such as, e.g., "concept", "organization" or "person".
+#' 
+#' @param object A dna_barplot object.
+#' @param lab.pos,lab.neg Names for (dis-)agreement labels.
+#' @param lab Should (dis-)agreement labels and title be displayed?
+#' @param colors If \code{TRUE}, attribute colors will be used to fill the
+#'   bars.
+#' @param fontSize Text size in pts.
+#' @param barWidth Thickness of the bars. Bars will touch when set to \code{1}.
+#'   When set to \code{0.5}, space between two bars is the same as thickness of
+#'   bars.
+#' @param axisWidth Thickness of the x-axis which separates agreement from
+#'   disagreement.
+#' @param truncate Sets the number of characters to which axis labels should be truncated.
+#' @param exclude.min Reduces the plot to entities with a minimum frequency of statements.
+#'
+#' @examples
+#' \dontrun{
+#' dna_init()
+#' dna_sample()
+#'
+#' dna_openDatabase("sample.dna", coderId = 1, coderPassword = "sample")
+#' 
+#' # compute barplot data
+#' b <- dna_barplot(statementType = "DNA Statement",
+#'                  variable = "concept",
+#'                  qualifier = "agreement")
+#' 
+#' # plot barplot with ggplot2                
+#' autoplot(b)
+#' }
+#'
+#' @author Johannes B. Gruber, Tim Henrichsen
+#'
+#' @seealso \link{dna_barplot}
+#' 
+#' @importFrom ggplot2 autoplot
+#' @importFrom ggplot2 ggplot
+#' @importFrom ggplot2 aes_string
+#' @importFrom ggplot2 geom_line
+#' @importFrom ggplot2 theme_minimal
+#' @importFrom ggplot2 theme
+#' @importFrom ggplot2 geom_bar
+#' @importFrom ggplot2 position_stack
+#' @importFrom ggplot2 coord_flip
+#' @importFrom ggplot2 element_blank
+#' @importFrom ggplot2 element_text
+#' @importFrom ggplot2 scale_color_identity
+#' @importFrom ggplot2 scale_fill_identity
+#' @importFrom ggplot2 geom_text
+#' @importFrom ggplot2 .pt
+#' @importFrom ggplot2 annotate
+#' @importFrom ggplot2 scale_x_discrete
+#' @export
+autoplot.dna_barplot <- function(object, 
+                                 lab.pos = "Agreement",
+                                 lab.neg = "Disagreement",
+                                 lab = TRUE,
+                                 colors = FALSE,
```

Comment:
I think it would make sense to let the user select attributes for the colour. `dna_barplot` adds all attribute variables as columns after the `int` values. I think a potentially good solution could be to allow the `colors` argument to have one of the following values (add sanity checks in the beginning!):

- `NULL` or `FALSE`: no colours.
- `TRUE` or `color`: use the RGB colour like now.
- character object matching one of the attribute variables: pick the attribute to color the bars.

By adding this last option, we could create barplots like the one we are using for the backbone paper more easily.

Reply:
I thought so too, but wouldn't that only make sense if the provided character object actually contains colours? If the user provides an attribute or an additional character object that doesn't contain any RGB colours, then the user would need to add two additional ggplot arguments (scale_color_manual, scale_fill_manual) to the plot, in which the colours are defined.

Wouldn't it be easier if the user changes the color column in the dna_barplot object before plotting it? I could add an example in the help file on how to do that.

Reply:
OK, good suggestion! Yes, add an example to the help page, then it should be fine. The example should take one of the attribute columns from the `dna_barplot` object. But do make sure you add those sanity checks and give appropriate error messages, as mentioned in the comment.